tests/rust/aligned.toml

```diff
@@ -0,0 +1,38 @@
+header = """
+#define CBINDGEN_PACKED        __attribute__ ((packed))
+#define CBINDGEN_UNION_PACKED  CBINDGEN_PACKED
+#define CBINDGEN_STRUCT_PACKED CBINDGEN_PACKED
+
+#define CBINDGEN_ALIGNED(n)     __attribute__ ((aligned(n)))
+#define CBINDGEN_UNION_ALIGNED  CBINDGEN_ALIGNED
+#define CBINDGEN_STRUCT_ALIGNED CBINDGEN_ALIGNED
+"""
+
+[union]
+packed = "CBINDGEN_UNION_PACKED"
+aligned_n = "CBINDGEN_UNION_ALIGNED"
```

Comment:
I think with GCC, they consistently use `aligned(...)` and `packed` -- `enum`s might be a bit odd but that's because you need to pass a compiler flag to enable all the alignment features.

Reply:
Ugh, MSVC only supports `#pragma` to pack stuff? That kinda sucks :(

Reply:
[Yeah](https://docs.microsoft.com/en-us/cpp/preprocessor/pack?view=vs-2019). Luckily MSVC does have attributes for alignment ([`alignas`](https://docs.microsoft.com/en-us/cpp/cpp/alignof-and-alignas-cpp?view=vs-2019) and [the older `__declspec(align(n))`](https://docs.microsoft.com/en-us/cpp/cpp/align-cpp?view=vs-2019)). So this is only an issue for packing.

But we could actually implement MSVC-style packing (as a separate option), because [GCC does actually support `#pragma pack`](https://gcc.gnu.org/onlinedocs/gcc-4.4.4/gcc/Structure_002dPacking-Pragmas.html). It would look something like:

```
#pragma pack(push, 1)
typedef struct foo {
  // ...
};
#pragma pack(pop)
```

Reply:
Yup, I concur. If somebody needs it it should be pretty easy to add support for it.